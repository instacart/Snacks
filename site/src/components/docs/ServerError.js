import React, {Fragment} from 'react'
import Playground from '../Playground'

export default () => (

ServerError Example:

<Playground>
{`
<div>
  <TextField
    name="test"
    type="email"
    floatingLabelText="Email"
    hintText="Enter your email address"
    serverError="The email you have entered is already taken"
    defaultValue="foobar@instacart.com"
    isValid={false}
  />
</div>
`}
</Playground>

)
