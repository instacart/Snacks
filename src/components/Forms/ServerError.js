import React, { Component } from 'react'
import PropTypes            from 'prop-types'
import Radium               from 'radium'
import { colors }           from '../../styles'

const styles = {
  root: {
    backgroundColor: 'rgba(233, 0, 56, 0.1)',
    borderRadius: '4px',
    color: colors.RED_500,
    fontSize: '15px',
    marginBottom: '12px',
    paddingTop: '15px',
    paddingRight: '15px',
    paddingBottom: '15px',
    paddingLeft: '15px'
  }
}

@Radium
class ServerError extends Component {
  static propTypes = {
    /** Override styles */
    style: PropTypes.object,
    /** Error text */
    text: PropTypes.string
  }

  render() {
    const {
      style,
      text
    } = this.props

    return (
      <div
        style={[styles.root, style]}
        aria-live={'assertive'}
        aria-atomic={true}
      >
        {text}
      </div>
    )
  }
}

export default ServerError
