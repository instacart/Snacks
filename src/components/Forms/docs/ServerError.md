ServerError Example:

```js
<div>
  <TextField
    name="test"
    type="email"
    floatingLabelText="Email"
    hintText="Enter your email address"
    serverError="The email you have entered is already taken"
    value="foobar@instacart.com"
    isValid={false}
  />
</div>
```
