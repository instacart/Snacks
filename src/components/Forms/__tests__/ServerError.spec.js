import React from 'react'
import renderer from 'react-test-renderer'
import ServerError from '../ServerError'

it('renders ServerError correctly', () => {
  const tree = renderer
    .create(
      <div>
        <ServerError text="Error Text" />
      </div>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
