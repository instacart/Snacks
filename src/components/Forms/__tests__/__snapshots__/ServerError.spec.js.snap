// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ServerError correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      aria-atomic={true}
      aria-live="assertive"
      data-radium={true}
      style={
        Object {
          "backgroundColor": "rgba(233, 0, 56, 0.1)",
          "borderRadius": "4px",
          "color": "#E6003D",
          "fontSize": "15px",
          "marginBottom": "12px",
          "padding": "15px",
        }
      }
    >
      Error Text
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
